packages/relay-compiler/transforms/RelayGenerateIDFieldTransform.js

```diff
@@ -49,7 +49,11 @@ type State = {
 function relayGenerateIDFieldTransform(
   context: CompilerContext,
 ): CompilerContext {
-  const idType = assertLeafType(context.serverSchema.getType(ID_TYPE));
+  const idType = context.serverSchema.getType(ID_TYPE);
+  if (!idType) {
+    throw new Error('Schema does not contain an id field on any types');
+  }
```

Comment:
I’m not sure I understand this. The original code appears to try to get the `ID` type, shouldn’t that be a builtin type that should always exist?

Regardless, can you use `invariant` here instead, which appears to be the way assertions are performed throughout the existing code-base?

Reply:
Ok, I should dive deeper into when the `ID` type actually exists, as there’s #2285 by @graingert that also mentions this issue.

Interestingly enough, @sibelius mentions that it would be fixed by my #2249 PR, although I didn’t knew of that issue at the time. @sibelius have you verified that? @brad-decker & @graingert would you mind verifying if #2249 fixes your issues and then please leave feedback on the PR? 🙏

Reply:
@alloy -- will do. Thanks for checking this out.

Reply:
Ve sakam srcki ste!! <333

Reply:
schema without ID that will throw this error https://github.com/freitasgst/weather-express-graphql/commit/c8ee408afc2e5ac4deaeb45d260800040199f57d

Reply:
@sibelius You indicated [here](https://github.com/facebook/relay/issues/2285#issuecomment-358309351) that #2249 fixes it; is that indeed the case or not after all?

Reply:
Not after all, this is the right fix for that